library/kani_macros/src/sysroot/contracts.rs

```diff
@@ -809,6 +904,40 @@ fn requires_ensures_main(attr: TokenStream, item: TokenStream, is_requires: bool
     output.into()
 }
 
+fn pats_to_idents<P>(
+    sig: &mut syn::punctuated::Punctuated<syn::FnArg, P>,
+) -> impl Iterator<Item = Ident> + '_ {
+    sig.iter_mut().enumerate().map(|(i, arg)| match arg {
+        syn::FnArg::Receiver(_) => Ident::new("self", Span::call_site()),
+        syn::FnArg::Typed(syn::PatType { pat, .. }) => {
+            let ident = Ident::new(&format!("arg{i}"), Span::mixed_site());
+            *pat.as_mut() = syn::Pat::Ident(syn::PatIdent {
+                attrs: vec![],
+                by_ref: None,
+                mutability: None,
+                ident: ident.clone(),
+                subpat: None,
+            });
+            ident
+        }
+    })
+}
+
+struct SelfDetector(bool);
+
+impl<'ast> Visit<'ast> for SelfDetector {
+    fn visit_path(&mut self, i: &'ast syn::Path) {
+        self.0 |= i.get_ident().map_or(false, |i| i == "self")
```

Comment:
Nit: use `SelfValue` and `SelfType`?

Reply:
Not possible (I'm fairly sure, I double checked). This is the canonical way to match for `self` as an expression, because `self` as an expression is (simplified) `Expr(PathExpr { path: Path { path: [Ident("self")] } })`. `SelfType` and `SelfValue` is only used in function signatures.

Reply:
This actually made me realize that we don't have to match for `self`, but can just check if there's a `Receiver`

Reply:
Your comments sent me down a bit of a rabbit hole wo I made some changes and actually added some tests.
I'm not super happy with the way I match for `Self`, the fact that the `Ident`s only compare equal with `Span::mixed_site()` makes me feel uneasy somehow. But I guess it's fine